### PR TITLE
refactor(core): cleanup module implementation

### PR DESCRIPTION
**Pull request by bartlomieju**
* remove "ModuleNameMap", instead define that map inline inside "Modules" struct
* remove "dyn_import_id" argument from "mod_instantiate"
* rename "Modules" struct to "ModuleMap"
* rename "JsRuntime::modules" to "JsRuntime::module_map"

More PRs to follow

### CODE REVIEW

File: core/modules.rs [hunk 1]
```diff
@@ -398,84 +398,46 @@ enum SymbolicModule {
...
+    let mut mod_name = name;
+    loop {
+      let symbolic_module = self.by_name.get(mod_name)?;
+      match symbolic_module {
+        SymbolicModule::Alias(target) => {
+          mod_name = target;
+        }
+        SymbolicModule::Mod(mod_id) => return Some(*mod_id),
+      }
+    }
```
Comment (ry) [inline core/modules.rs:432]: Maybe there should be a counter here to detect infinite loops?

doc string would be helpful for this function.

File: core/runtime.rs [hunk 2]
```diff
@@ -734,58 +734,46 @@ impl JsRuntime {
   /// `AnyError` can be downcast to a type that exposes additional information
   /// about the V8 exception. By default this type is `JsError`, however it may
   /// be a different type if `RuntimeOptions::js_error_create_fn` has been set.
-  fn mod_instantiate(
-    &mut self,
-    id: ModuleId,
-    dyn_import_id: Option<ModuleLoadId>,
-  ) -> Result<(), AnyError> {
```
Comment (ry) [inline core/runtime.rs:741]: Nice

File: core/modules.rs (hunk 1 above)
Comment (bartlomieju) [reply core/modules.rs:432]: I don't think so, this function hasn't changed in a long time and works as expected. Added a docstring.

**Review by ry — APPROVED** [review]
LGTM - nice clean up

**Pull request merged**